src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -246,13 +247,55 @@ protected override void VisitSwitchSection(BoundSwitchSection node, bool isLastS
             VisitStatementList(node);
         }
 
+        private struct PossiblyConditionalState
+        {
+            public LocalState State;
```

Comment:
randomly, I wonder if this field can be collapsed into `StateWhenTrue`, and accessed via properties which assert on the value of `IsConditionalState`. No need to try and change things around to do it, just a thought that entered my mind.

Reply:
I agree this is tempting. I'm not sure why AbstractFlowPass doesn't do something like that. 
But given the existing pattern, I prefer to stay with separate fields. It's more consistent and possibly clearer. This is something I hope we could make easier with discriminated union types.

---
In reply to: [573999711](https://github.com/dotnet/roslyn/pull/51001#discussion_r573999711) [](ancestors = 573999711)